### PR TITLE
Upgrade to k8s 1.27

### PR DESCRIPTION
**Pull request by tomleb**
## Problem

We want to upgrade client-go and stuff to 1.27 release.

## Solution

Upgrade a few dependencies. Notably:
- `controller-runtime` to 0.15.0: This version is needed for support for 1.27.
- `github.com/rancher/lasso/controller-runtime`: This version is needed to support controller-runtime 0.15.
- `k8s.io/*`: Most of them to `v0.27.4`.
- `github.com/rancher/client-go` to v1.27.4-rancher1.

## CheckList

I have done the following manual tests:

- Made sure the webhook still works by deleting the ValidatingWebhookConfiguration and restarting `webhook`. It recreated it as expected.
- Created a new `Restricted Admin` user and verified that it can't update the `Admin` role.
- ~~I did not test the helm chart deployment, that still needs to be done.~~ I have tested uninstalling and reinstalling the chart via `helm uninstall` and `helm install`. This worked fine as well.

### CODE REVIEW

**Review by MbolotSuse — COMMENTED** [review]
LGTM, will leave final approval if/when we can get CI Passing.

**Review by KevinJoiner — CHANGES_REQUESTED** [review]
Can you update the go version to 1.20 in go.mod
as well as the version [dockerfile.Dapper](https://github.com/rancher/webhook/blob/ffa666dba12fcf1fbdc333ae73eb45025dafac9e/Dockerfile.dapper#L1) file

**Review by MbolotSuse — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,146 +1,149 @@
 module github.com/rancher/webhook
 
-go 1.19
+go 1.20
 
 // on release remove this wrangler replace and use the latest tag
 replace github.com/rancher/wrangler v1.1.1 => github.com/rancher/wrangler v1.1.1-0.20230705223603-201b4da5bdaf
```
Comment (KevinJoiner) [inline go.mod:6]: Can you also update wrangler to this `github.com/rancher/wrangler v1.1.1-0.20230831050635-df1bd5aae9df`

File: pkg/resources/management.cattle.io/v3/project/validator.go [hunk 2]
```diff
@@ -177,7 +179,18 @@ func usedQuotaFits(usedQuota, projectQuota *v3.ResourceQuotaLimit) (*field.Error
...
 	fits, exceeded := quotaFits(usedQuotaResourceList, projectQuotaResourceList)
 	if !fits {
-		return field.Forbidden(projectSpecFieldPath.Child(projectQuotaField), fmt.Sprintf("resourceQuota is below the used limit on fields: %s", format.ResourceList(exceeded))), nil
+		return field.Forbidden(projectSpecFieldPath.Child(projectQuotaField), fmt.Sprintf("resourceQuota is below the used limit on fields: %s", formatResourceList(exceeded))), nil
 	}
 	return nil, nil
 }
+
+// directly copied from https://github.com/kubernetes/kubernetes/blob/a66aad2d80dacc70025f95a8f97d2549ebd3208c/pkg/kubelet/util/format/resources.go
+func formatResourceList(resources v1.ResourceList) string {
```
Comment (tomleb) [inline pkg/resources/management.cattle.io/v3/project/validator.go:188]: Missed the k8s.io/kubernetes upgrade somehow. The `format.ResourceList` function was removed some time ago in the upstream project. I'm not sure what's the policy on copying (keep attribution somehow, etc etc?) and also I see that the rancher/rancher project is taking the same approach here: https://github.com/rancher/rancher/pull/42691. So I'm wondering, where should that function live? Do we want a common place for it or do we keep it copied in both?

Comment (KevinJoiner) [reply pkg/resources/management.cattle.io/v3/project/validator.go:188]: This function is straightforward and has no critical usage, so I think we can follow the go proverb.
> [A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

**Review by KevinJoiner — APPROVED** [review]

**Review by MbolotSuse — APPROVED** [review]

**Pull request merged**